pkg/collector/corechecks/servicediscovery/module/envs.go

```diff
@@ -0,0 +1,99 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
```

Comment:
shouldn't this file be accessible only for linux?

Reply:
I've kept it the same as `impl_linux.go`, which doesn't have that.

Reply:
if the file name ends with `_linux.go` (or `_windows.go`) it implies `linux` build only

Reply:
This reflects in the job failure https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/603129477

Reply:
TIL. Done.